File: Dialogue/DialogSet.cpp

#include <vector>
#include <string>
#include <sstream>
#include <cstdlib>
#include "./DialogSet.h"

//This fails build, dunno why.
//It tells me there is a redefinition of "DialogSet"
//WHERE WHERE THE FUCK IS REDEFINITION YOU PEACE OF...
//Time wasted: 15 hours. (I'm not joking)

DialogSet::DialogSet(std::vector<std::string> &lines) : Displayable(lines)
{
    for (std::string line : lines)
    {
        dialogLines.push_back(line);
    }
}
Displayable DialogSet::getRandomDialog()
{
    randomIndex = rand() % dialogLines.size();
    return Displayable(dialogLines[randomIndex]);
}

File: TavernSharp/UiManagement/DialogSet.cs



public class DialogSet : Displayable {
    private String[] dialogLines;
    private Random random;

    public DialogSet(String[] lines){
        dialogLines = lines;
        random = new Random();
    }

    public Displayable getRandomDialog() {
        int randomIndex = random.Next(dialogLines.Length);
        return new Displayable(dialogLines[randomIndex]);
        //This will not work correctly, issue is in Displayable types i guess?`
    }
}